packages/checkout-storefront/src/sections/PaymentSection/AdyenDropIn/useAdyenDropin.ts

```diff
@@ -0,0 +1,321 @@
+
+          if (!transaction || !adyenData) {
+            return;
+          }
+
+          if (adyenData) {
+            void handlePaymentResult({
+              paymentResponse: adyenData.paymentResponse,
+              transaction,
+            });
+          }
+        },
+      }),
+      [
+        adyenCheckoutSubmitParams?.component,
+        commonErrorMessages.somethingWentWrong,
+        handlePaymentResult,
+        showCustomErrors,
+        transactionInitialize,
+      ]
+    )
+  );
+
+  const onTransactionProccess = useSubmit<
+    TransactionProcessMutationVariables,
+    typeof transactionProccess
+  >(
+    useMemo(
+      () => ({
+        onSubmit: transactionProccess,
+        onError: () => {
+          // will tell the processing screen to disappear
+          setIsProcessingPayment(false);
+          // we don't do these at onFinished since redirect will happen first
+          clearQueryParams("transaction");
+          setCurrentTransactionId(null);
+
+          showCustomErrors([{ message: commonErrorMessages.somethingWentWrong }]);
+          adyenCheckoutSubmitParams?.component.setStatus("ready");
+        },
+        extractCustomErrors: (result) => result?.data?.transactionProcess?.data?.errors,
+        onSuccess: ({ data }) => {
+          if (!data?.data) {
+            showCustomErrors([{ message: commonErrorMessages.somethingWentWrong }]);
+            return;
+          }
+
+          const {
+            transaction,
+            data: { paymentDetailsResponse },
+          } = data;
+
+          // we don't do these at onFinished since redirect will happen first
+          clearQueryParams("transaction");
+          setCurrentTransactionId(null);
+
+          handlePaymentResult({
+            paymentResponse: paymentDetailsResponse,
+            transaction,
+          });
+        },
+      }),
+      [
+        adyenCheckoutSubmitParams?.component,
+        commonErrorMessages.somethingWentWrong,
+        handlePaymentResult,
+        setIsProcessingPayment,
+        showCustomErrors,
+        transactionProccess,
+      ]
+    )
+  );
+
+  const onSubmitInitialize: AdyenCheckoutInstanceOnSubmit = useEvent(async (state, component) => {
+    component.setStatus("loading");
+    setAdyenCheckoutSubmitParams({ state, component });
+    validateAllForms(authenticated);
+    setSubmitInProgress(true);
+  });
+
+  useEffect(() => {
+    const validating = anyFormsValidating(validationState);
+    const allFormsValid = areAllFormsValid(validationState);
+
+    // any of the conditions below - do nothing
+    if (!submitInProgress || validating || anyRequestsInProgress || !adyenCheckoutSubmitParams) {
+      return;
+    }
+
+    // there was en error either in some other request or form validation
+    // - stop the submission altogether
+    if (!finishedApiChangesWithNoError || !allFormsValid) {
+      adyenCheckoutSubmitParams?.component.setStatus("ready");
+      setSubmitInProgress(false);
```

Comment:
```suggestion
```

Remove it since we're always doing this (see above comment)

Reply:
I mean not always - the first condition keeps it as it is when entering so I moved it after the first condition